### ISSUE TITLE
[Details List]: Accessibility MacOS Table properties are not defined correctly with VoiceOver app in Mac

### ISSUE THREAD

**Issue by caradong** [OfficeDev/office-ui-fabric-react#4747]
### Bug Report

- __Package version(s)__: 5.70.0
- __Browser and OS versions__: Safari Version: 11.0.3 (12604.5.6.1.1)  MacOS Sierra: 10.12.6

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Accessibility

Products/sites affected: Details List

#### Describe the issue:

Using VoiceOver app on a Mac, with Safari, multiple table properties are not correct.

#### Actual behavior:

·         Table dimensions are defined incorrectly. 
·         User is unable to interact with the table using ‘Ctrl + Option + Shift + Up/Down arrow’. (in case of VoiceOver only). On pressing ‘Ctrl + Option + Shift + Down arrow’, VO announces as ‘In table No selection. 1 column, 5 rows row 0 of 5 selected’
·         On pressing up/down arrow keys VoiceOve announces the entire row information.

#### Expected behavior:

·         Table dimensions should be defined correctly.
·         User should be able to interact with the table using ‘Ctrl + Option + Shift + Up/Down arrow’.
·         On interacting with the table, user should be able to navigate to the individual data items using ‘Ctrl + Option + arrow keys’ and the data items should be read out with the associated column header and the appropriate row and column number.

**Comment by ThomasMichon** [comment]
The table dimensions should have been fixed via #5066. @caradong it's worth testing again.

**Comment by mikewheaton** [comment]
I've verified that this still repros. In Edge and Chrome each row gets focus and the full contents are read out:
![image](https://user-images.githubusercontent.com/1382445/47182026-5b855100-d2d9-11e8-9e48-6d8c52319629.png)

Whereas in Safari, focus falls on specific table cells (reading 'blank' in the case of an icon) and the keyboard shortcuts that Safari reads out don't work. Neither do the standard arrow keys. It's very difficult to navigate:
![image](https://user-images.githubusercontent.com/1382445/47182073-722ba800-d2d9-11e8-8b12-919219ae914e.png)

**Comment by natalieethell** [comment]
Hey @caradong thanks for opening this issue! @mikewheaton I just tested out VoiceOver in Safari on DetailsList, and I was able to successfully use Control + Option + Up/Down Arrow and just Up/Down arrows to navigate the list. VoiceOver in Safari does seem to focus on the first cell instead of the entire row, but I don't think that's a behavior that Fabric can fix.

### **Incorrect Table Dimensions Read Out**

I also noticed a table dimension issue in VoiceOver in Safari that I didn't see in VoiceOver in Chrome (or any other screen reader + browser). It's reading out more columns than are actually in the DetailsList on the demo page.

When navigating to the checkbox on the [Compact DetailsList demo page](https://developer.microsoft.com/en-us/fabric#/components/detailslist/compact), it reads out the incorrect number of columns (it should be 3):
**Toggle selection, unchecked, checkbox, table 4 columns, ...**

I tried rendering this exact DetailsList in a [CodePen](https://codepen.io/naethell/pen/qQbYRM), and the number of columns (3) is read out correctly for some reason.
**Toggle selection, unchecked, checkbox, table 3 columns, ...**

I think that, for some reason, on the demo page VoiceOver in Safari is grouping an element above the DetailsList with the columns. It seems to do this for other DetailsList examples with elements above them as well. 

The number of columns, however, is correctly set on the element with role="grid".

![image](https://user-images.githubusercontent.com/14134000/48171917-b693dd80-e2b2-11e8-88f8-c87c4834f565.png)

**Comment by natalieethell** [comment]
So, my theory is that there's something on the demo page that's causing the VoiceOver in Safari to think there are more columns than there actually are, but I'm not sure what it is.

**Comment by micahgodbolt** [comment]
Seems to be an issue with demo page? If this is working in production can we close the issue and create a new one to capture docs work?
